cmd/build/build.sh

```
@@ -206,14 +263,107 @@ fi
+# So a bad value fails RIGHT HERE with a message that names the template key and
+# what to do about it. Note what this does NOT do: it does not default, coerce or
+# tolerate the bad value. `unset` remains the ONE accepted absence, and it is an
+# explicit sentinel a caller writes on purpose (a hand-run build.sh outside
+# cmd/build), never something produced by a missing input.
+validate_artifact_provenance() {
+    local name="$1" value="$2" template_key="$3" pattern="$4" shape="$5"
+    if [ "${value}" = "unset" ]; then
+        return 0
+    fi
+    case "${value}" in
+        $pattern) return 0 ;;
```

Shellcheck (0.11.0)

[warning] Quote expansions in case patterns to match literally rather than as a glob.

(SC2254)